src/com/liakhovskyi/ArrayList.java

```diff
@@ -11,51 +11,70 @@ public ArrayList() {
         elements = new Object[DEFAULT_CAPACITY];
     }
 
-    public ArrayList(int size) {
-        elements = new Object[size];
+    public ArrayList(int capacity) {
+        elements = new Object[capacity];
     }
 
     @Override
     public T get(int index) {
-        if (index >= elements.length && index < 0) {
-            throw new ArrayIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
-        }
+        checkForIndexException(index);
         return (T) elements[index];
     }
 
     @Override
     public void add(T t) {
-        if (size == elements.length) {
-            increaseSize();
-        }
+        increaseSize();
         elements[size] = t;
         size++;
     }
 
     @Override
     public void remove(int index) {
-        if (index >= 0 && index < size) {
-            for (int i = index; i < size; i++) {
-                elements[i] = elements[i + 1];
-            }
-            size--;
-            if (elements.length > DEFAULT_CAPACITY && size == elements.length / 4) {
-                decreaseSize();
-            }
+        checkForIndexException(index);
+        checkForEmptyListException();
+
+        for (int i = index; i < size; i++) {
+            elements[i] = elements[i + 1];
         }
+        size--;
+        decreaseSize();
     }
 
     @Override
     public int size() {
         return size;
     }
 
+    private void checkForIndexException(int index) {
+        if (index >= size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+    }
+
+    private void checkForEmptyListException() {
+        if (isEmpty()) {
```

Comment:
this is not a correct check
delete this method
if you have empty list and try get or remove some element your "checkForIndexException" will throw a correct exception